larky/src/main/java/com/verygood/security/larky/modules/vgs/vault/DefaultVault.java

```diff
@@ -0,0 +1,184 @@
+        String sValue = getValue(value);
+        Object secret = getStorage(storage).get(sValue);
+        return secret == null ? "token" : secret; // return 'token' if entry not found
+    }
+
+    private String getValue(Object value) throws EvalException {
+        if (!(value instanceof String)) {
+            throw Starlark.errorf(String.format(
+                    "Value of type %s is not supported in DefaultVault, expecting String", value.getClass().getName()
+            ));
+        }
+
+        return value.toString();
+    }
+
+    private Map<String, Object> getStorage(Object storage) throws EvalException {
+
+        if (storage instanceof NoneType) { // Use 'persistent` storage by default
+            return persistentVaultStorage;
+        } else if (storage instanceof String) {
+            if (!storageConfig.containsKey(storage)) {
+                throw Starlark.errorf(String.format(
+                        "Storage '%s' not found in available storage list [persistent, volatile]", storage
+                ));
+            }
+
+            return storageConfig.get(storage);
+        }
+
+        throw Starlark.errorf(String.format(
+                "Storage of type %s is not supported in DefaultVault, expecting String",
+                storage.getClass().getName()
+        ));
+    }
+
+    private AliasGenerator getTokenizer(Object format) throws EvalException {
+
+        if (format instanceof NoneType) {
+            return formatTokenizer.get("UUID"); // default tokenizer
+        } else if (format instanceof String) {
+            if (formatTokenizer.containsKey(format)) {
+                return formatTokenizer.get(format);
+            } else {
+                throw Starlark.errorf(String.format(
+                        "Format '%s' not found", format
+                ));
+            }
+        }
+
+        throw Starlark.errorf(String.format(
+                "Format of type %s is not supported in DefaultVault, expecting String",
+                format.getClass().getName()
+        ));
+    }
+
+    private class NumberLengthPreserving extends ValidatingAliasGenerator {
+
+        private final Pattern cardPattern = Pattern.compile("\\d{3,16}");
+
+        @Override
+        protected String internalTokenize(String value) {
+            return RandomStringUtils.randomNumeric(value.length());
+        }
+
+        @Override
+        protected boolean isValid(String value) {
+            return cardPattern.matcher(value).find();
+        }
+
+        @Override
+        protected AliasGenerator fallbackAliasGenerator() {
+            return new RawAliasGenerator();
+        }
+    }
+
+    private abstract class ValidatingAliasGenerator implements AliasGenerator {
+
+        @Override
+        public String tokenize(String value) throws EvalException {
+            if (!isValid(value)) {
+                return fallbackAliasGenerator().tokenize(value);
+            }
+            return internalTokenize(value);
```

Comment:
I know that this follows logic from Satellit, but I whink we need to reassess this.\ in this PR and in Satellite
This looks like something that goes agains our logic on proxy.
@rozenbom why do you have a fallback generator for value in Satellite if redactable data is now valid according to the format?

Reply:
@aslepakurov becase this is ho the proxy behaves and I tried to replicate it. https://verygoodsecurity.atlassian.net/wiki/spaces/EN/pages/33031585/Alias+formats